src/main/kotlin/org/rust/lang/core/parser/RustParserUtil.kt

```diff
@@ -343,9 +360,17 @@ object RustParserUtil : GeneratedParserUtilBase() {
         tokenType == RsElementTypes.MACRO_EXPR &&
             '}' == b.originalText.subSequence(startOffset, endOffset).findLast { it == '}' || it == ']' || it == ')' }
 
-    private fun contextualKeyword(b: PsiBuilder, keyword: String, elementType: IElementType): Boolean {
+    private fun contextualKeyword(
+        b: PsiBuilder,
+        keyword: String,
+        elementType: IElementType,
+        requiredNextElements: TokenSet = DEFAULT_REQUIRED_NEXT_ELEMENTS,
+        invertRequiredNextElements: Boolean = true
```

Comment:
According to the prev comment, we should find a better name for this flag too but currently, I can come up with nothing suitable